src/HPolyhedron.jl

```diff
@@ -499,6 +501,43 @@ function remove_redundant_constraints!(P::PT;
     return P
 end
 
+"""
+    linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+
+Concrete linear map of a polyhedron in constraint representation.
+
+### Input
+
+- `M` -- matrix
+- `P` -- polyhedron in constraint representation
+
+### Output
+
+A polyhedron of the same type as the input (`PT`).
+
+### Algorithm
+
+If the matrix ``M`` is invertible (which we check with a sufficient condition),
+then ``y = M x`` implies ``x = \\text{inv}(M) y`` and we transform the
+constraint system ``A x ≤ b`` to ``A \\text{inv}(M) y ≤ b``.
+"""
+function linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+    if !isinvertible_sufficient(M)
+        if P isa HPolyhedron
+            error("linear maps for polyhedra need to be invertible")
+        end
+        # use the implementation for general polytopes
+        return invoke(linear_map, Tuple{typeof(M), AbstractPolytope{N}}, M, P)
+    end
+    # matrix is invertible
+    invM = inv(M)
+    constraints = Vector{LinearConstraint{N}}(undef, length(constraints_list(P)))
+    for c in constraints_list(P)
+        push!(constraints, LinearConstraint(vec(c.a' * invM), c.b))
```

Comment:
probably more or less the same efficiency but shorter:

```julia
A, b = tosimplehrep(P)
return PT(A * invM, b)
```

Reply:
Cool, but it needs to create two additional matrices. So I keep mine :man_shrugging: